scripts/migrations/generate.sh

```
@@ -17,8 +17,26 @@ PGUSER_="${PGUSER_:-flexprice}"; PGPASS_="${PGPASS_:-flexprice123}"
+if ! make generate-ent >/dev/null 2>&1; then
+  echo "FAIL: 'make generate-ent' failed — fix ent/schema/ first" >&2
+  exit 1
+fi
+
 psql "$BASE/postgres?sslmode=disable" -q -c "DROP DATABASE IF EXISTS mig_draft;" \
                                      -c "CREATE DATABASE mig_draft;" >/dev/null
+# Build the scratch database the way a FRESH INSTALL is built: baseline snapshot
+# first, then the versioned migrations. The baseline lives outside the timeline —
+# dbmate never runs it — so applying only the migrations leaves an empty schema and
+# Ent proposes creating everything.
+BASELINE="$(ls -1 migrations/baseline/postgres_baseline_ent_*.sql 2>/dev/null | tail -1)"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)